data-serving/data-service/src/controllers/preprocessor.ts

```diff
@@ -128,3 +135,27 @@ export const createCaseRevision = async (
 
     next();
 };
+
+export const createBatchCaseRevisions = async (
+    request: Request,
+    response: Response,
+    next: NextFunction,
+) => {
+    const casesToUpsert = (await findCasesWithCaseReferenceData(request)).map(
+        (c) => {
+            return {
+                case: c,
+            };
+        },
+    );
+
+    await CaseRevision.insertMany(casesToUpsert, {
```

Comment:
Yeah, it _feels like_ there's a MongoDB operation that prevents us from placing these cases into memory by chaining these two operations, but perhaps not worth sweating too much over.

Reply:
I agree. :\ I looked for a `copy` or `clone` but couldn't find anything. But there very well could be something out there that I didn't find -- either way, there's definitely room for improvement. 

I filed #784 for this, but let me add that as something to investigate further

Reply:
All good, thanks!